content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,260 @@ The following are not considered as part of the adversarial model:
 In specific, the communication channels are assumed to be secure.
 
 # Wire Specification
-Peers communicate with each other using a request / response API. 
-The messages sent are Protobuf RPC messages which are implemented using [protocol buffers v3](https://developers.google.com/protocol-buffers/).
-The following are the specifications of the Protobuf messages.
 
 ## Payloads
 
 ```protobuf
 syntax = "proto3";
 
-message Index {
-  bytes digest = 1;
-  sint64 receiverTime = 2;
-  sint64 senderTime = 3;
-  string pubsubTopic = 4;
-}
-
-message PagingInfo {
-  uint64 pageSize = 1;
-  Index cursor = 2;
-  enum Direction {
-    BACKWARD = 0;
-    FORWARD = 1;
-  }
-  Direction direction = 3;
-}
+// Protocol identifier: /vac/waku/store/3.0.0
+package waku.store.v3;
 
-message ContentFilter {
-  string contentTopic = 1;
-}
+import "waku/message/v1/message.proto";
 
-message HistoryQuery {
-  // the first field is reserved for future use
-  string pubsubtopic = 2;
-  repeated ContentFilter contentFilters = 3;
-  PagingInfo pagingInfo = 4;
+message WakuMessageKeyValue {
+  optional bytes message_hash = 1; // Globally unique key for a Waku Message
+  optional waku.message.v1.WakuMessage message = 2; // Full message content as value
 }
 
-message HistoryResponse {
-  // the first field is reserved for future use
-  repeated WakuMessage messages = 2;
-  PagingInfo pagingInfo = 3;
-  enum Error {
-    NONE = 0;
-    INVALID_CURSOR = 1;
-  }
-  Error error = 4;
+message StoreRequest {
```

Comment:
Would it be better to name is `StoreQuery` as we are actually quering the Store and not requesting to Store data?

Reply:
No strong opinion, but trying to start being more consistent in our Request-Response protocols by being explicit about the `...Request` and `...Response` messages (see Filter as an example). You could argue that we are sending the wire protocol _request_ to the store node to perform a query, though this boils down to semantics.

Reply:
Agreed, but when you read `StoreRequest` it immediately means that we are Requesting information to be stored. 
Better to name it `StoreQueryRequest` then.

Reply:
Right. Which would indeed be a rename of the protocol to "Store Query" protocol as you suggested elsewhere.